Zend/zend_compile.c

```diff
@@ -2867,8 +2865,33 @@ static void zend_compile_list_assign(
 
 		zend_verify_list_assign_target(var_ast, old_style);
 
-		zend_emit_op(&fetch_result, ZEND_FETCH_LIST, expr_node, &dim_node);
-		zend_emit_assign_znode(var_ast, &fetch_result);
+		opline = zend_emit_op(&fetch_result, ZEND_FETCH_LIST, expr_node, &dim_node);
+		if (elem_ast->attr) {
+			opline->extended_value = ZEND_LIST_MAKE_WRITABLE;
+			prev_op = (opline - 1);
+			do {
+				if (prev_op->opcode == ZEND_FETCH_LIST) {
+					if (prev_op->op1.var == opline->op1.var) {
+						continue;
+					} else if (prev_op->result.var == opline->op1.var) {
+						if (prev_op->extended_value == ZEND_LIST_MAKE_WRITABLE) {
+							break;
+						}
+						prev_op->extended_value = ZEND_LIST_MAKE_WRITABLE;
+						opline = prev_op;
+						continue;
+					}
+					break;
+				} else if (prev_op->opcode == ZEND_ASSIGN_REF || prev_op->opcode == ZEND_ASSIGN) {
```

Comment:
Hmm, yes.  I'm just trying to limit how far back we need to look.  Obviously this is for optimization purposes here, but since it's during compile and not execute, I guess I could drop that check and just keep looking back through this op-stack.  Not exactly efficient, but not sure if that's better or worse than keeping looking back through the op-array